### PR TITLE
Add guide for Integrating AMP into your app

### PR DESCRIPTION
**Pull request by bpaduch**
Adding new guide for Integrating AMP pages with your app.  This PR is implementing the content that was already written and reviewed (in the Google Doc) into Markdown format for the ampproject site.

Preview of what it'll look like: [Here](https://barb-ampdocs.firebaseapp.com/docs/guides/integrate-with-apps.html)

to: @sebastianbenz 
cc: @cramforce, @rudygalfi, @pbakaus, @adewale 

Fixes: #548

### CODE REVIEW

File: content/docs/guides/integrate-with-apps.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
+---
+$title: Integrate AMP with your app 
+$order: 12
+$category: Develop
+toc: true
+
+---
+
+This guide is intended for app developers who want to integrate AMP pages into
+their app.
```
Comment (sebastianbenz) [inline content/docs/guides/integrate-with-apps.md]: Maybe explain a bit when this is useful, for example: 'This guide is intended for developers who want to link to the AMP version of content surfaced in their mobile or web app, for example, a mobile chat application could show the fast loading AMP version of a shared URL to provide better user experience.' (this still sounds a bit rough)

Comment (bpaduch) [reply content/docs/guides/integrate-with-apps.md]: Good point.  As this guide has information on AMP viewer, I hesitate saying it's just for linking. How about:

This guide is intended for mobile and web app developers who want to integrate and link to AMP pages. For example, consider a mobile chat app that loads the AMP version of a shared URL to achieve a faster experience for users.

Comment (sebastianbenz) [reply content/docs/guides/integrate-with-apps.md]: Way better :-)

**Comment by sebastianbenz** [comment]
LGTM

**Comment by bpaduch** [comment]
Thanks @sebastianbenz!

**Pull request merged**